modules/@apostrophecms/login/index.js

```diff
@@ -981,6 +994,77 @@ module.exports = {
             required: true
           })
           );
+      },
+
+      normalizeLoginName(usernameOrEmail, {
+        caseInsensitive = self.options.caseInsensitive
+      } = {}) {
+        if (typeof usernameOrEmail !== 'string' || !caseInsensitive) {
+          return usernameOrEmail;
+        }
+        return usernameOrEmail.toLowerCase();
+      },
+
+      async addCaseInsensitiveMigration() {
+        if (self.options.caseInsensitive) {
+          self.apos.migration.add('login-case-insensitive', self.caseInsensitiveTask);
+        }
+      },
+
+      async caseInsensitiveTask() {
+        const duplicatedUsernames = [];
+        await self.apos.migration.eachDoc({ type: '@apostrophecms/user' }, 1, async (user) => {
+          const normalizedUsername = self.apos.login
+            .normalizeLoginName(user.username, { caseInsensitive: true });
+          const normalizedEmail = self.apos.login
+            .normalizeLoginName(user.email, { caseInsensitive: true });
+
+          const shouldUpdateUsername = user.username !== normalizedUsername;
+          const shouldUpdateEmail = user.email && user.email !== normalizedEmail;
+          if (!shouldUpdateUsername && !shouldUpdateEmail) {
+            return;
+          }
+
+          try {
+            if (shouldUpdateUsername) {
+              await self.apos.user.safe.updateOne(
+                { _id: user._id },
+                {
+                  $set: {
+                    username: normalizedUsername
+                  }
+                }
+              );
+            }
+            await self.apos.doc.db.updateOne(
+              { _id: user._id },
+              {
+                $set: {
+                  username: normalizedUsername,
+                  ...shouldUpdateEmail && { email: normalizedEmail }
+                }
+              }
+            );
+          } catch (err) {
+            if (self.apos.doc.isUniqueError(err)) {
+              duplicatedUsernames.push({
+                _id: user._id,
+                username: user.username,
+                email: user.email
+              });
+              return;
+            }
+            throw err;
+          }
+        });
+
+        if (duplicatedUsernames.length) {
+          self.logError(
+            'conflicting-usernames',
+            'Some usernames changed in lowercase already exist for other users, please fix it or they won\'t be able to log in anymore',
```

Comment:
It might be the username, it might be the email. I think that's OK but we need to express that ambiguity to the user running the task.